### PR TITLE
Use getters instead of direct references that may have been uninitialized. #2951

### LINKED ISSUE #2951: Potential use of uninitialized variables in DSOIIdentifierProvider
**Describe the bug**
`#getDOIOutOfObject(DSpaceObject)` makes direct use of the fields `PREFIX` and `NAMESPACE_SEPARATOR`.  But these seeming constants are in fact filled from DSpace configuration by internal getters `getPrefix()` and `getNamespaceSeparator()`, and may be null if directly used.

**To Reproduce**
I tripped over this when running the single test while working on an unrelated issue.

**Expected behavior**
The test should have succeeded but instead asserted a mismatch with the expected value.

### PR DESCRIPTION
**Pull request by mwoodiupui**
## References
* Fixes #2951

## Description
Use provided getters for lazy-initialized fields, instead of direct reference which may find the fields uninitialized.

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* Use getPrefix() and getNamespaceSeparator() instead of PREFIX and NAMESPACE_SEPARATOR.
* Hoist construction of unvarying string out of comparison loop.
* Remove un-necessary explicit boxing of `char` constants; let the compiler handle it efficiently.

Tests should pass, as before.  I ran into this when running a single test from the suite.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by mwoodiupui** [comment]
Please ignore mention of NPE in commit comment.  That was not what happened.

**Review by tdonohue — APPROVED** [review]
👍 Yep, this an an obvious bug.  Both PREFIX and NAMESPACE_SEPARATOR might be uninitialized here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java#L94-L105

You are correct, we need to be calling their getters to ensure this code always works right.

Will wait for Travis CI to approve (via the tests).  After that, I'm merging with 1 approval, as this is so tiny.

**Pull request merged**
